app/models/normalize_primo_record.rb

```diff
@@ -107,13 +107,55 @@ def links
       links << { 'url' => record_link, 'kind' => 'full record' }
     end
 
-    # Add openurl if available
-    links << { 'url' => openurl, 'kind' => 'openurl' } if openurl.present?
+    # If $$Topenurl_article
+    if @record['pnx']['links'] && @record['pnx']['links']['openurl'] && openurl.present?
+      links << { 'url' => openurl, 'kind' => 'Check Availability' }
+    end
+
+    # Add PDF if available
+    if @record['pnx']['links'] && @record['pnx']['links']['linktopdf']
+
+      parsed_string = parse_link_string(@record['pnx']['links']['linktopdf'].first)
+
+      if parsed_string['U'].present?
+        links << { 'url' => parsed_string['U'],
+                   'kind' => 'View PDF' }
+      end
+    end
+
+    # Add HTML if available
+    if @record['pnx']['links'] && @record['pnx']['links']['linktohtml']
+
+      parsed_string = parse_link_string(@record['pnx']['links']['linktohtml'].first)
+
+      if parsed_string['U'].present?
+        links << { 'url' => parsed_string['U'],
+                   'kind' => 'View HTML' }
+      end
+    end
 
     # Return links if we found any
     links.any? ? links : []
   end
 
+  # Parses a link string into a hash of key-value pairs.
+  # The link string is a series of key-value pairs separated by $$, where each pair is prefixed by a single character.
+  # For example: "$$Uhttps://example.com$$TView PDF" would be parsed into { 'U' => 'https://example.com', 'T' => 'View PDF' }
```

Comment:
Is this construct documented anywhere? How stable is it? My assumption is that this isn't something new, but I'm unfamiliar with it as a format.

I like having a method that will decompose a single string like this into its component parts, and agree with not trying to pluck out values in this method - that happens in the context of whatever logic is fishing for something.